app/assets/javascripts/check-for-edits-before-submit.js

```diff
@@ -0,0 +1,18 @@
+(function ($) {
+  'use strict';
+
+  var form = $('.js-check-for-edits-before-submit');
+  var form_original_data = form.serialize();
+
+  $(".js-before-submit").on('click submit', function() {
+    if (form.serialize() != form_original_data) {
+      if(!confirm('You have unsaved changes to the current month.\n\nProceed and discard changed?')) {
```

Comment:
@benbarnett  Not sure the new code is as readable but covers as many edge cases as I could get it to. might be worth going through it together if you can see ways of tidying it up.